### ISSUE TITLE
Statevector output wrong on GPU

### ISSUE THREAD

**Issue by scarrazza** [Qiskit/qiskit-aer#1319]
### Informations

- **Qiskit Aer version**: 0.8.2 (qiskit-aer-gpu)
- **Python version**: 3.8.10
- **Operating system**: Linux Ubuntu 20.04.2 LTS
- **CUDA Version**:  11.4 (from ubuntu repository)
- **CUDA driver**: 470.57.02
- **GPU model**: Quadro T2000, Titan V, A6000, V100

### What is the current behavior?

The statevector simulation on GPU returns wrong final states if the number of qubits is >= 10.

### Steps to reproduce the problem

Using the following minimal example:
```python
#!/usr/bin/env python
from qiskit.providers.aer import StatevectorSimulator
from qiskit.circuit.library import QFT
from qiskit import transpile

c = QFT(26)
simulator = StatevectorSimulator(method='statevector_gpu')
tc = transpile(c, simulator)
r = simulator.run(tc).result()
print(r.get_statevector(tc))
```
I get the |0> state instead of the QFT with 26 qubits state vector:
```
[1.+0.j 0.+0.j 0.+0.j ... 0.+0.j 0.+0.j 0.+0.j]
```
Moreover, if I add a repetition loop this state returns random solutions.

### What is the expected behavior?

The exact QFT final state. I have also noticed that for small number of qubits the GPU is always on idle, like a sort of CPU fallback, however I can't find information about that behaviour in the documentation.

**Comment by chriseclectic** [comment]
@doichanj Can you look into this?

**Comment by chriseclectic** [comment]
Should be fixed by #1325
